libcore/FileOperations/CommonJob.vala

```diff
@@ -64,4 +64,184 @@ public class Files.FileOperations.CommonJob {
     protected bool aborted () {
         return cancellable.is_cancelled ();
     }
+
+    protected void abort_job () {
+        cancellable.cancel ();
+    }
+
+    protected void skip_file (GLib.File file) {
+        if (skip_files == null) {
+            skip_files = new GLib.GenericSet<GLib.File> (GLib.File.hash, GLib.File.equal);
+        }
+
+        skip_files.add (file);
+    }
+
+    protected void skip_readdir_error (GLib.File dir) {
+        if (skip_readdir_error_set == null) {
+            skip_readdir_error_set = new GLib.GenericSet<GLib.File> (GLib.File.hash, GLib.File.equal);
+        }
+
+        skip_readdir_error_set.add (dir);
+    }
+
+    protected bool should_skip_file (GLib.File file) {
+        if (skip_files != null) {
+            return skip_files.contains (file);
+        }
+
+        return false;
+    }
+
+    protected bool should_skip_readdir_error (GLib.File dir) {
+        if (skip_readdir_error_set != null) {
+            return skip_readdir_error_set.contains (dir);
+        }
+
+        return false;
+    }
+
+    protected void verify_destination (GLib.File dest, out string? dest_fs_id, int64 required_size) {
+        dest_fs_id = null;
+        try {
+            var info = dest.query_info (GLib.FileAttribute.STANDARD_TYPE + "," +
+                                        GLib.FileAttribute.ID_FILESYSTEM,
+                                        GLib.FileQueryInfoFlags.NONE,
+                                        cancellable);
+            var file_type = info.get_file_type ();
+            dest_fs_id = info.get_attribute_string (GLib.FileAttribute.ID_FILESYSTEM);
+
+            if (file_type != GLib.FileType.DIRECTORY) {
+                var dest_name = dest.get_parse_name ();
+                /// TRANSLATORS: '\"%s\"' is a placeholder for the quoted basename of a file.  It may change position but must not be translated or removed
+                /// '\"' is an escaped quoted mark.  This may be replaced with another suitable character (escaped if necessary)
+                string primary = _("Error while copying to \"%s\".").printf (dest_name);
+                unowned string secondary = _("The destination is not a folder.");
```

Comment:
Why is the secondary string unowned?  It does not seem to belong to anything else.

Reply:
because it is a static string in the code, it is equivalent in C to:
```c
const char *secondary = "my static string";
```
while in Vala:
```vala
string secondary = "my string";
```
is translated in C to:
```c
char *secondary = g_strdup ("my string");
[…]
g_free (secondary);
```
So it means one less string allocation

Reply:
I see, thanks for the explanation.